pkg/volume/csi/csi_mounter.go

```diff
@@ -172,28 +228,27 @@ func (c *csiMountMgr) SetUpAt(dir string, fsGroup *int64) error {
 		return err
 	}
 	if podAttrs != nil {
-		if attribs == nil {
-			attribs = podAttrs
+		if volAttribs == nil {
+			volAttribs = podAttrs
 		} else {
 			for k, v := range podAttrs {
-				attribs[k] = v
+				volAttribs[k] = v
 			}
 		}
 	}
 
-	fsType := csiSource.FSType
 	err = csi.NodePublishVolume(
 		ctx,
-		c.volumeID,
-		c.readOnly,
+		volumeHandle,
```

Comment:
the way we're initializing empty variables, then setting variables in giant switch blocks up above, then using them down here makes me uneasy.

It is hard to tell which of these variables is *allowed* to be `nil` or `empty`, which are `required` and whether they are actually set at this point. I don't really have a solution to this but I wanted to flag it in case you could think of a better way.

@msau42 have any ideas on how to deal with this?

Reply:
Would it help if we defined some internal type, and then had translation methods from PVSource/InlineSource -> internal type?

Reply:
I think that would be an elegant solution.
@vladimirvivien I think we're running low on time so lets not solve it in this PR but lets create an issue for it and link it in a TODO comment.